packages/core/src/baseclient.ts

```diff
@@ -794,11 +795,11 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
       .then(processedEvent => {
         if (processedEvent === null) {
           this.recordDroppedEvent('before_send', dataCategory, event);
-          if (isTransactionEvent(event)) {
+          if (isTransaction) {
             const spans = event.spans || [];
             // the transaction itself counts as one span, plus all the child spans that are added
             const spanCount = 1 + spans.length;
-            this._outcomes['span'] = (this._outcomes['span'] || 0) + spanCount;
```

Comment:
Noticed that this was actually buggy, because the key is built from reason + category 😬 now this is unified and should work consistently.